src/legacy/status_im/log_level/core.cljs

```diff
@@ -1,51 +1,24 @@
 (ns legacy.status-im.log-level.core
   (:require
     [legacy.status-im.multiaccounts.update.core :as multiaccounts.update]
-    [re-frame.core :as re-frame]
-    [taoensso.timbre :as log]
-    [utils.i18n :as i18n]
+    [native-module.core :as native-module]
     [utils.re-frame :as rf]))
 
 (rf/defn save-log-level
   {:events [:log-level.ui/change-log-level-confirmed]}
-  [{:keys [db]} log-level]
+  [{:keys [db] :as cofx} log-level]
   (let [old-log-level (get-in db [:profile/profile :log-level])]
     (when (not= old-log-level log-level)
       (let [need-set-log-enabled? (or (empty? old-log-level) (empty? log-level))
-            log-enabled?          (boolean (seq log-level))
-            rpc-calls             (cond-> []
-                                    log-enabled?
-                                    (conj {:method   "wakuext_setLogLevel"
-                                           :params   [{:logLevel log-level}]
-                                           :on-error #(log/error "Failed to set log level" %)})
+            log-enabled?          (boolean (seq log-level))]
+        (when log-enabled?
+          (native-module/set-log-level log-level))
```

Comment:
@qfrank events shouldn't have side-effects. You would need to create effect(s) and use them here within `:fx`, somewhat similar to the original code.